cpp/src/arrow/compute/benchmark-util.h

```diff
@@ -70,5 +70,28 @@ void RegressionSetArgs(benchmark::internal::Benchmark* bench) {
   BenchmarkSetArgsWithSizes(bench, {kL1Size});
 }
 
+// RAII struct to handle some of the boilerplate in regression benchmarks
+struct RegressionArgs {
+  // size of memory tested (per iteration) in bytes
+  const int64_t size;
+
+  // proportion of nulls in generated arrays
+  const double null_proportion;
+
+  explicit RegressionArgs(benchmark::State& state)
+      : size(state.range(0) / 4),
```

Comment:
What is `state.range(0)` and why divide it by 4?

Reply:
This is copied from [compare-benchmark.cc](https://github.com/apache/arrow/pull/4366/files/5a224b24ced09ed6692e0d0ef73a399dab072229#diff-c04bee063fbd2d6922621b436a47dd81R33) and now that I look at it I think it's an error in both locations; state.range(0) should be the byte size of the memory tested and it's used that way in [aggregate-benchmark.cc](https://github.com/bkietz/arrow/blob/5a224b24ced09ed6692e0d0ef73a399dab072229/cpp/src/arrow/compute/kernels/aggregate-benchmark.cc#L311). @fsaintjacques can you confirm?